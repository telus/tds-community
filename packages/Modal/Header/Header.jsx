import React from 'react'
import PropTypes from 'prop-types'
import Heading from '@tds/core-heading'
import Box from '@tds/core-box'
import Text from '@tds/core-text'

import { HEADER_LEVELS, TEXT_SIZES } from '../configs'

const Header = ({ heading, headingLevel, subHeading, subHeadingSize }) => {
  return (
    <Box between={3} tabIndex={0}>
      <Box between={3}>
        <Heading level={headingLevel}>{heading}</Heading>
        {subHeading && <Text size={subHeadingSize}>{subHeading}</Text>}
      </Box>
    </Box>
  )
}

Header.propTypes = {
  heading: PropTypes.string.isRequired,
  headingLevel: PropTypes.oneOf(Object.values(HEADER_LEVELS)),
  subHeading: PropTypes.string,
  subHeadingSize: PropTypes.oneOf(Object.values(TEXT_SIZES)),
}

Header.defaultProps = {
  headingLevel: HEADER_LEVELS.H3,
  subHeading: '',
  subHeadingSize: TEXT_SIZES.MEDIUM,
}

export default Header
