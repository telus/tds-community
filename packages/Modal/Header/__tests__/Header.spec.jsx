import React from 'react'
import { mount } from 'enzyme'

import Header from '../Header'

describe('Header', () => {
  const doMount = (props = {}) => mount(<Header heading="Mock text" {...props} />)

  it('renders', () => {
    const header = doMount()

    expect(header).toMatchSnapshot()
  })
})
