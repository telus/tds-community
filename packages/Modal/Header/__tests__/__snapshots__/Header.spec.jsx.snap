// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
.c1 {
  word-wrap: break-word;
  color: #2a2c2e;
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.4;
  -webkit-letter-spacing: -0.6px;
  -moz-letter-spacing: -0.6px;
  -ms-letter-spacing: -0.6px;
  letter-spacing: -0.6px;
}

.c1 sup {
  position: relative;
  vertical-align: baseline;
  padding-left: 0.1em;
  font-size: 0.875rem;
  top: -0.5em;
}

.c1 > span {
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:not(:last-child) {
  margin-bottom: 1rem;
}

<Header
  heading="Mock text"
  headingLevel="h3"
  subHeading=""
  subHeadingSize="medium"
>
  <Box
    between={3}
    inline={false}
    tabIndex={0}
    tag="div"
  >
    <styled.div
      between={3}
      inline={false}
      tabIndex={0}
      tag="div"
    >
      <StyledComponent
        between={3}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        inline={false}
        tabIndex={0}
        tag="div"
      >
        <div
          className="c0"
          tabIndex={0}
        >
          <Box
            between={3}
            inline={false}
            tag="div"
          >
            <styled.div
              between={3}
              inline={false}
              tag="div"
            >
              <StyledComponent
                between={3}
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      [Function],
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c0",
                      "rules": Array [
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                inline={false}
                tag="div"
              >
                <div
                  className="c0"
                >
                  <Heading
                    invert={false}
                    level="h3"
                  >
                    <Heading__StyledHeading
                      as="h3"
                      data-testid="heading"
                      invert={false}
                      level="h3"
                    >
                      <StyledComponent
                        as="h3"
                        data-testid="heading"
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "Heading__StyledHeading-mpz92r-0",
                              "isStatic": false,
                              "lastClassName": "c1",
                              "rules": Array [
                                "word-wrap: break-word;",
                                [Function],
                              ],
                            },
                            "displayName": "Heading__StyledHeading",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "Heading__StyledHeading-mpz92r-0",
                            "target": "h1",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        invert={false}
                        level="h3"
                      >
                        <h3
                          className="c1"
                          data-testid="heading"
                        >
                          Mock text
                        </h3>
                      </StyledComponent>
                    </Heading__StyledHeading>
                  </Heading>
                </div>
              </StyledComponent>
            </styled.div>
          </Box>
        </div>
      </StyledComponent>
    </styled.div>
  </Box>
</Header>
`;
